core/src/test/java/org/apache/calcite/rex/RexProgramTest.java

```diff
@@ -2458,6 +2458,79 @@ private SqlOperator getNoDeterministicOperator() {
     checkSimplifyUnchanged(isNull(cast(vVarcharNotNull(), tInt(false))));
   }
 
+  /** Unit test for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5769">[CALCITE-5769]
+   * Optimizing 'CAST(e AS t) IS NOT NULL' to 'e IS NOT NULL'</a>. */
+  @Test void testSimplifyCastIsNull3() {
+    // "(cast A as bigint) IS NULL" when A is int and A is not null
+    // ==>
+    // "false"
+    checkSimplify(isNull(cast(vIntNotNull(), tBigInt(false))), "false");
```

Comment:
I'm a little confused about this `vIntNotNull()`. Per its java doc, it returns a non-nullable int variable, however, in `vParamNotNull()`, whose java doc says this method returns nullable varchar variable.

Reply:
Looks like the java doc has some problem. We can fix it in another issue.